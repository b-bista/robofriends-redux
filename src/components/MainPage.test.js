import { shallow, mount, render} from 'enzyme';
import React from 'react';
import MainPage from './MainPage';

let wrapper;

beforeEach(() => {
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots: [],
        searchField: '',
        isPending: false
    }
    wrapper = shallow(<MainPage { ...mockProps }/>);
})

it('tests that main page is rendered', () => {
    expect(wrapper).toMatchSnapshot();
})

it('filters robots correctly', () => {
    const mockProps2 = {
        onRequestRobots: jest.fn(),
        robots: [
            {
                id: 3,
                name: 'josh',
                email: 'josh@gmail.com'
            },
            {
                id: 4,
                name: 'jake',
                email: 'jake@gmail.com'
            }
        ],
        searchField: 'jo',
        isPending: false
    }
    const wrapper2 = shallow(<MainPage { ...mockProps2 }/>);
    expect.assertions(1);
    expect(wrapper2.instance().filterRobots()).toEqual([{
        id: 3,
        name: 'josh',
        email: 'josh@gmail.com'
    }]);
})
