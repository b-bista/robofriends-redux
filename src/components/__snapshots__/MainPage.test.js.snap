// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests that main page is rendered 1`] = `ShallowWrapper {}`;
